pkg/storage/chunk/cache/resultscache/cache_test.go

```diff
@@ -682,6 +682,107 @@ func Test_resultsCache_MissingData(t *testing.T) {
 	require.False(t, hit)
 }
 
+func Test_shouldCacheReq(t *testing.T) {
+	modelNow := model.Now()
+
+	cfg := Config{
+		CacheConfig: cache.Config{
+			Cache: cache.NewMockCache(),
+		},
+	}
+	c, err := cache.New(cfg.CacheConfig, nil, log.NewNopLogger(), stats.ResultCache, constants.Loki)
+	require.NoError(t, err)
+	rc := NewResultsCache(
+		log.NewNopLogger(),
+		c,
+		nil,
+		ConstSplitter(day),
+		mockLimits{},
+		MockMerger{},
+		MockExtractor{},
+		nil,
+		nil,
+		func(_ context.Context, tenantIDs []string, r Request) int {
+			return 10
+		},
+		nil,
+		false,
+		false,
+	)
+	require.NoError(t, err)
+
+	// create cache with handler
+	ctx := user.InjectOrgID(context.Background(), "1")
+
+	// create request with start end within the key extents
+	req := parsedRequest.WithStartEndForCache(time.UnixMilli(int64(modelNow)-(50*1e3)), time.UnixMilli(int64(modelNow)-(10*1e3)))
+
+	// fill cache
+	key := ConstSplitter(day).GenerateCacheKey(context.Background(), "1", req)
+	rc.put(ctx, key, []Extent{mkExtent(int64(modelNow)-(600*1e3), int64(modelNow))})
+
+	// Asserts (when `shouldCacheReq` is non-nil and set to return false (should not cache), resultcache should get result from upstream handler (mockHandler))
+	// 1. With `shouldCacheReq` non-nil and set `noCacheReq`, should get result from `next` handler
```

Comment:
should we change this `shouldCacheReq` to `shouldLookupCache` or `bypassCache`?
the current name is a bit confusing, it reads "whether this request should be cached"?

Reply:
Agree. I can change it in my tests. But the original type still called `ShouldCacheReq` (IMO, still confusing, because the same request's results can still be cached after getting it from the upstream).

Should ideally be called something like `ShouldLookupCache`.  I will fix the naming on my test now. Renaming the original type I can do it in separate PR may be.